src/validateOPATCatalogValues.ts

```diff
@@ -0,0 +1,99 @@
+// src/validateOPATCatalogValues.ts
+
+import { ValidatorResult } from "./ValidatorResult";
+import { validateCatalog } from "./validateCatalog";
+
+export function validateOPATCatalogValues(
+  data: any,
+  catalog: any
+): ValidatorResult {
+  const catalogSchema = "opat-catalog-0.1.0.json";
+  const validCatalogResult = validateCatalog(catalog, catalogSchema);
+  let validationPassed = true;
+  const validationMessages = [];
+
+  if (validCatalogResult.result) {
+    if (catalog.chapters) {
+      const catalogChapters = catalog.chapters;
+
+      for (const catalogChapter of catalogChapters) {
+        if (data.chapters[catalogChapter.id].criteria) {
+          const dataCriteriaList = data.chapters[catalogChapter.id].criteria;
+          for (const dataCriteria of dataCriteriaList) {
+            if (
+              !checkForNumInCatalogChapters(
+                dataCriteria.num,
+                catalogChapter.id,
+                catalogChapters
+              )
+            ) {
+              validationPassed = false;
+              validationMessages.push(
+                `criteria num '${dataCriteria.num}' is not included in '${catalogChapter.label}'`
+              );
+            }
+
+            if (dataCriteria.components) {
+              const dataComponents = dataCriteria.components;
+              if (catalog.components) {
+                for (const dataComponent of dataComponents) {
```

Comment:
If we add component per-criteria (as suggested above) we will need to revise this to look at the criteria specific list.